### PR TITLE
FYST-526 Update refund taxes owed UI and add financial information to MD pdf/xml

### PR DESCRIPTION
**Pull request by arinchoi03**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-526
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Removed `bank_name` from all intakes!
- Deconstructed account_holder_name for MD, since the XML requires the names separated out
- Added `joint_account_holder_name` fields for MD
- Implemented follow up question component for joint account holder name ([see thread for convo if curious](https://cfastaff.slack.com/archives/C07GN6YB93N/p1732217503719199))
- MD xml/pdf information flow through from tax_refund/taxes_owed -- lot of the xml is already filled out via [another ticket](https://codeforamerica.atlassian.net/browse/FYST-1088); see subtasks for specifics needed to be implemented in this ticket

## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Unit/integration/manual tests
### UI
Screenshot 2024-11-22 at 12 53 46 PM

### XML
Screenshot 2024-11-24 at 12 14 59 AM
Screenshot 2024-11-24 at 12 15 07 AM

### PDF
Screenshot 2024-11-24 at 12 15 27 AM

### CODE REVIEW

**Comment by arinchoi03** [comment]
still need to implement validations that match with schema, i just copied over what was in the name dob form at the moment.

File: app/views/state_file/questions/md_tax_refund/_md_bank_details.html.erb [hunk 1]
```diff
@@ -34,13 +33,29 @@
...
+      <%= form.cfa_input_field :account_holder_first_name, "First Name", classes: ["form-width--long"] %>
+      <%= form.cfa_input_field :account_holder_middle_initial, "Middle Initial", classes: ["form-width--long"] %>
+      <%= form.cfa_input_field :account_holder_last_name, "Last Name", classes: ["form-width--long"] %>
+      <%= form.cfa_input_field :account_holder_suffix, "Suffix", classes: ["form-width--long"] %>
+      <div class="question-with-follow-up__follow-up same-card-as-question" id="joint-account-holder-questions">
+        <p>Joint Account Holder Name</p>
+        <%= form.cfa_input_field :joint_account_holder_first_name, "First Name", classes: ["form-width--long"] %>
+        <%= form.cfa_input_field :joint_account_holder_middle_initial, "Middle Initial", classes: ["form-width--long"] %>
+        <%= form.cfa_input_field :joint_account_holder_last_name, "Last Name", classes: ["form-width--long"] %>
+        <%= form.cfa_input_field :joint_account_holder_suffix, "Suffix", classes: ["form-width--long"] %>
```
Comment (arinchoi03) [inline app/views/state_file/questions/md_tax_refund/_md_bank_details.html.erb:50]: need translations for all the labels -- a lot of them already should exist

File: app/lib/efile/md/md502_calculator.rb [hunk 2]
```diff
@@ -420,6 +422,28 @@ def calculate_line_40
...
+        else
+          full_name
+        end
+      end
+
+      def full_name(is_joint: false)
+        first = @intake.send("#{is_joint ? 'joint_' : ""}account_holder_first_name")
+        middle_initial = @intake.send("#{is_joint ? 'joint_' : ""}account_holder_middle_initial")
+        last = @intake.send("#{is_joint ? 'joint_' : ""}account_holder_last_name")
+        suffix = @intake.send("#{is_joint ? 'joint_' : ""}account_holder_suffix")
```
Comment (arinchoi03) [inline app/lib/efile/md/md502_calculator.rb:442]: idk if this is better or worse

File: app/lib/pdf_filler/md502_pdf.rb [hunk 3]
```diff
@@ -86,13 +86,24 @@ def hash_for_pdf
         'Enter 3': @xml_document.at('Form502 Additions StateRetirementPickup')&.text,
         'Enter 6': @xml_document.at('Form502 Additions Total')&.text,
         'Enter 7': @xml_document.at('Form502 Additions FedAGIAndStateAdditions')&.text,
-        "Enter 15": @xml_document.at('Form502 Subtractions Total')&.text,
-        "Enter 16": @xml_document.at('Form502 Subtractions StateAdjustedGrossIncome')&.text,
+        'Enter 15': @xml_document.at('Form502 Subtractions Total')&.text,
+        'Enter 16': @xml_document.at('Form502 Subtractions StateAdjustedGrossIncome')&.text,
```
Comment (arinchoi03) [inline app/lib/pdf_filler/md502_pdf.rb:90]: rubocop told me to make it single quotes 👮‍♂️

File: app/models/state_file_az_intake.rb [hunk 4]
```diff
@@ -86,7 +85,7 @@
 #  index_state_file_az_intakes_on_spouse_state_id_id   (spouse_state_id_id)
 #
 class StateFileAzIntake < StateFileBaseIntake
-  self.ignored_columns = %w[charitable_cash charitable_noncash household_excise_credit_claimed_amt tribal_wages armed_forces_wages]
+  self.ignored_columns = %w[charitable_cash charitable_noncash household_excise_credit_claimed_amt tribal_wages armed_forces_wages bank_name]
```
Comment (arinchoi03) [inline app/models/state_file_az_intake.rb:88]: not related, but I think i have to add `:df_data_import_failed` here since it's not being removed from the annotations

File: app/lib/pdf_filler/md502_pdf.rb [hunk 5]
```diff
@@ -131,6 +143,26 @@ def generate_codes_for_502_su
...
     end
 
+    def full_names_of_bank_account_holders
+      return nil unless @intake.payment_or_deposit_type.to_sym == :direct_deposit
+
+      if @intake.has_joint_account_holder_yes?
+        account_holder_full_name + " and " + account_holder_full_name(for_joint: true)
+      else
+        account_holder_full_name
+      end
```
Comment (arinchoi03) [inline app/lib/pdf_filler/md502_pdf.rb:153]: generally we pull information from the XML -- not sure if I should pull from the XML using `xml_document.at("FirstName")` etc instead of directly calling these methods on the intake?

File: app/models/state_file_az_intake.rb [hunk 6]
```diff
@@ -86,7 +85,7 @@
 #  index_state_file_az_intakes_on_spouse_state_id_id   (spouse_state_id_id)
 #
 class StateFileAzIntake < StateFileBaseIntake
-  self.ignored_columns = %w[charitable_cash charitable_noncash household_excise_credit_claimed_amt tribal_wages armed_forces_wages]
+  self.ignored_columns = %w[charitable_cash charitable_noncash household_excise_credit_claimed_amt tribal_wages armed_forces_wages bank_name df_data_import_failed]
```
Comment (anisharamnani) [inline app/models/state_file_az_intake.rb:88]: Why did you add `df_data_import_failed` as well in this list of ignored columns?

File: app/models/state_file_base_intake.rb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 class StateFileBaseIntake < ApplicationRecord
-  self.ignored_columns = [:df_data_import_failed]
+  self.ignored_columns = [:df_data_import_failed, :bank_name]
```
Comment (anisharamnani) [inline app/models/state_file_base_intake.rb:2]: if we add bank_name to the shared base intake then we don't need to add it for Az right?

File: spec/forms/state_file/md_tax_refund_form_spec.rb [hunk 8]
```diff
@@ -61,8 +67,15 @@
...
-          account_holder_name: "Laney Knope"
+          account_holder_first_name: "Laney",
+          account_holder_middle_initial: "",
+          account_holder_last_name: "Knope",
+          account_holder_suffix: "",
+          joint_account_holder_first_name: "Janey",
+          joint_account_holder_middle_initial: "",
+          joint_account_holder_last_name: "Knope",
+          joint_account_holder_suffix: "",
+          has_joint_account_holder: 'yes',
```
Comment (jenny-heath) [inline spec/forms/state_file/md_tax_refund_form_spec.rb:78]: is this info required for the spec? i can't seem to find where it's used

File: spec/forms/state_file/md_tax_refund_form_spec.rb [hunk 9]
```diff
@@ -123,11 +156,59 @@
...
           expect(form).not_to be_valid
-          expect(form.errors).to include :account_holder_name
+          expect(form.errors).to include :account_holder_first_name
+          expect(form.errors).to include :account_holder_last_name
+        end
+      end
+
+      context "missing joint account holder name" do
+        let(:joint_account_holder_first_name) { nil }
+        let(:joint_account_holder_last_name) { nil }
```
Comment (jenny-heath) [inline spec/forms/state_file/md_tax_refund_form_spec.rb:171]: it looks like you're testing the following cases for these params:
- missing but not required 
- required and valid
- required and invalid 

but not testing the case where they're required and missing

File: app/assets/stylesheets/_state-file.scss [hunk 10]
```diff
@@ -287,7 +287,7 @@
       padding: 0 1rem;
     }
 
-    .question-with-follow-up__follow-up {
+    .question-with-follow-up__follow-up:not(.same-card-as-question) {
```
Comment (anisharamnani) [inline app/assets/stylesheets/_state-file.scss:290]: Sorry a little confused about what this is doing, would you mind explaining?

**Review by anisharamnani — APPROVED** [review]
@embarnard and i paired on reviewing this. 

⚠️ there is one pebble level comment where there is a translation missing. 

the rest of the comments & questions were just to get a better understanding of what's going on. 

thanks @arinchoi03 !

File: app/lib/efile/md/md502_calculator.rb [hunk 11]
```diff
@@ -420,6 +421,10 @@ def calculate_line_40
           @intake.state_file1099_rs.sum { |item| item.state_tax_withheld_amount.round }
       end
 
+      def calculate_authorize_direct_deposit
+        @intake.bank_authorization_confirmed_yes? ? "X" : nil
+      end
```
Comment (jenny-heath) [inline app/lib/efile/md/md502_calculator.rb:426]: would love to discuss the following, probably on tuple:

i know that [the original intent of the calculators was for them to mainly hold values that are numbers or booleans used to calculate other lines](https://github.com/codeforamerica/vita-min/pull/3689#issuecomment-1774152046). this doesn't look like it's used downstream anywhere so it could just be determined from the xml builder and left out of the calculator. either way i think i'd prefer not to have a calculator method/line set to the string "X" and instead return a boolean that can be converted to "X" in the xml builder (because "X" is required by an xml type). hopefully that makes sense!

Comment (jenny-heath) [reply app/lib/efile/md/md502_calculator.rb:426]: hm, update: i went and looked around and i guess this calculator is littered with methods that return "X" lol. i can't think of a reason this functionally matters so idk if i'm going to recommend a refactor? again, we can discuss

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 12]
```diff
@@ -446,5 +446,57 @@
...
+
+    context "Line 51d: NameOnBankAccount" do
+      before do
+        intake.payment_or_deposit_type = "direct_deposit"
+        intake.account_holder_first_name = "Jack"
+        intake.account_holder_middle_initial = "D"
+        intake.account_holder_last_name = "Hansel"
+      end
+
+      it 'outputs the total state and local tax withheld' do
```
Comment (jenny-heath) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:468]: oops this looks like a name, not tax withheld

File: app/lib/submission_builder/ty2024/states/md/documents/md502.rb [hunk 13]
```diff
@@ -140,6 +140,23 @@ def document
         end
       end
       xml.TaxWithheld calculated_fields.fetch(:MD502_LINE_40)
+      add_element_if_present(xml, "AuthToDirectDepositInd", :MD502_AUTHORIZE_DIRECT_DEPOSIT)
+      if @intake.payment_or_deposit_type.to_sym == :direct_deposit
```
Comment (jenny-heath) [inline app/lib/submission_builder/ty2024/states/md/documents/md502.rb:144]: i feel a bit confused that the joint account holder info has nothing to do with the payment/deposit type? is there a case where the type is _not_ direct deposit but there _is_ a joint account holder?

File: spec/lib/pdf_filler/md502_pdf_spec.rb [hunk 14]
```diff
@@ -455,5 +455,49 @@
...
+        allow_any_instance_of(Efile::Md::Md502Calculator).to receive(:calculate_authorize_direct_deposit).and_return 'X'
+      end
+
+      it "checks the refund information with the account holder's full name" do
+        expect(pdf_fields["Check Box 39"]).to eq "Yes"
+        expect(pdf_fields["Text Box 95"]).to eq "Jack D Hansel"
+        expect(pdf_fields['Check Box 41']).to eq "Yes"
+        expect(pdf_fields['Check Box 42']).to eq "Off"
+        expect(pdf_fields['Text Box 93']).to eq "123456789"
+        expect(pdf_fields['Text Box 94']).to eq "87654321"
```
Comment (jenny-heath) [inline spec/lib/pdf_filler/md502_pdf_spec.rb:481]: nit: inconsistent quotes

File: app/models/state_file_az_intake.rb (hunk 6 above)
Comment (jenny-heath) [reply app/models/state_file_az_intake.rb:88]: the one we actually use is called `:df_data_import_failed_at`. i tried to figure out the provenance of `:df_data_import_failed` but couldn't really get anywhere with it? [it seems like there was a move to replace `failed_at`](https://github.com/codeforamerica/vita-min/commit/0a2e8640995bb181cc7a5b2cc75524f574bd4a90). but i'm not sure how `:df_data_import_failed` was introduced...

File: app/models/state_file_base_intake.rb (hunk 7 above)
Comment (jenny-heath) [reply app/models/state_file_base_intake.rb:2]: ^ i think so too

File: app/models/state_file_md_intake.rb [hunk 15]
```diff
@@ -96,6 +104,7 @@
 #  index_state_file_md_intakes_on_spouse_state_id_id   (spouse_state_id_id)
 #
 class StateFileMdIntake < StateFileBaseIntake
+  self.ignored_columns = [:df_data_import_failed, :account_holder_name, :bank_name]
```
Comment (jenny-heath) [inline app/models/state_file_md_intake.rb:107]: same thing from other comments about `df_data_import_failed` and having `bank_name` on base intake

File: app/lib/efile/md/md502_calculator.rb (hunk 11 above)
Comment (anisharamnani) [reply app/lib/efile/md/md502_calculator.rb:426]: i asked em the same question when we were pairing on review. i was also curious because there's nothing being calculated here. em had said sometimes these methods are used elsewhere in the calculator for actual calculations. that doesn't seem to be the case here. 

👀 i'm curious what y'all decide post tuple & discussion!

File: db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb [hunk 16]
```diff
@@ -0,0 +1,14 @@
+class AddJointAccountHolderNameAndAccountHolderDeconstructedFields < ActiveRecord::Migration[7.1]
+  def change
+    add_column :state_file_md_intakes, :has_joint_account_holder, :integer
```
Comment (jenny-heath) [inline db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb:3]: give this a default of unfilled (0)

File: app/forms/state_file/md_tax_refund_form.rb [hunk 17]
```diff
@@ -6,16 +6,29 @@ class MdTaxRefundForm < QuestionsForm
...
+    SUFFIX_OPTIONS = %w[JR SR I II III IV V VI VII VIII IX X]
 
-      validates :bank_name, presence: true
+    with_options unless: -> { payment_or_deposit_type == "mail" } do
+      validates :account_holder_first_name, presence: true
+      validates :account_holder_last_name, presence: true
+      validates :account_holder_first_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,15})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.first_name.invalid_format') }}
+      validates :account_holder_last_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,31})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.last_name.invalid_format') }}
+      validates :account_holder_middle_initial, length: { maximum: 1 }, format: { with: /\A[A-Za-z]\z/.freeze, allow_blank: true }
+      validates :account_holder_suffix, inclusion: { in: SUFFIX_OPTIONS }, allow_blank: true, if: -> { joint_account_holder_suffix.present? }
```
Comment (jenny-heath) [inline app/forms/state_file/md_tax_refund_form.rb:31]: looks like these validations aren't tested?

File: app/forms/state_file/md_tax_refund_form.rb [hunk 18]
```diff
@@ -24,9 +37,20 @@ class MdTaxRefundForm < QuestionsForm
       validates :routing_number, presence: true, confirmation: true, routing_number: true
       validates :routing_number_confirmation, presence: true
 
+      validates :bank_authorization_confirmed, acceptance: { accept: 'yes', message: ->(_object, _data) { I18n.t("state_file.questions.md_tax_refund.md_bank_details.bank_authorization_confirmation_error") }}
```
Comment (jenny-heath) [inline app/forms/state_file/md_tax_refund_form.rb:40]: i don't see anything in the form spec about this field

File: app/lib/submission_builder/ty2024/states/md/documents/md502.rb (hunk 13 above)
Comment (arinchoi03) [reply app/lib/submission_builder/ty2024/states/md/documents/md502.rb:144]: oh you're right, it needs to be nested under this too

File: app/models/state_file_az_intake.rb (hunk 6 above)
Comment (arinchoi03) [reply app/models/state_file_az_intake.rb:88]: this is a typo... 😓

File: app/models/state_file_base_intake.rb (hunk 7 above)
Comment (arinchoi03) [reply app/models/state_file_base_intake.rb:2]: unfortunately, the `ignored_columns` get overwritten by the AZIntakes version. I can refactor to use the existing ignored columns and _add_ the other attributes being ignored in AZ

File: spec/forms/state_file/md_tax_refund_form_spec.rb (hunk 8 above)
Comment (arinchoi03) [reply spec/forms/state_file/md_tax_refund_form_spec.rb:78]: I think it's more that some info exists and then it's nulled out when the `payment_or_deposit_type` is `"mail"`-- which remidns me I should just fill every field with _something_

File: app/assets/stylesheets/_state-file.scss (hunk 10 above)
Comment (arinchoi03) [reply app/assets/stylesheets/_state-file.scss:290]: yea sorry, this is unclear.. basically the `#joint-account-holder-questions` element gets this `margin: 0` styling which makes it try to be centered within the container unintentionally. Is there a different way I could achieve this without making it be confusing? I thought `same-card-as-question` made sense at the time of implementation, but i guess maybe the difference here is just simply "not-centered"?

Screenshot 2024-11-25 at 1 20 22 PM

**another element with same styling**
Screenshot 2024-11-25 at 1 21 51 PM

Comment (arinchoi03) [reply app/assets/stylesheets/_state-file.scss]: @anisharamnani here's the clean up https://github.com/codeforamerica/vita-min/pull/5007/commits/1380324cb89f17fb7e3e1f9f832968ea6677ecc1

File: app/forms/state_file/md_tax_refund_form.rb [hunk 19]
```diff
@@ -6,16 +6,29 @@ class MdTaxRefundForm < QuestionsForm
...
-    with_options unless: -> { payment_or_deposit_type == "mail" } do
-      validates :account_holder_name, presence: true
+    SUFFIX_OPTIONS = %w[JR SR I II III IV V VI VII VIII IX X]
 
-      validates :bank_name, presence: true
+    with_options unless: -> { payment_or_deposit_type == "mail" } do
+      validates :account_holder_first_name, presence: true
+      validates :account_holder_last_name, presence: true
+      validates :account_holder_first_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,15})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.first_name.invalid_format') }}
+      validates :account_holder_last_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,31})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.last_name.invalid_format') }}
```
Comment (arinchoi03) [inline app/forms/state_file/md_tax_refund_form.rb:29]: ask Tiffany about following schema strictly? (allowing numbers in first/last name)

File: app/lib/efile/md/md502_calculator.rb (hunk 11 above)
Comment (arinchoi03) [reply app/lib/efile/md/md502_calculator.rb:426]: upon discussion: let's make this return a boolean & in the XML set `X` or `nil`

File: app/models/state_file_base_intake.rb (hunk 7 above)
Comment (arinchoi03) [reply app/models/state_file_base_intake.rb]: refactor: https://github.com/codeforamerica/vita-min/pull/5007/commits/b2219bd9849498259336b95b2a598e0680be77c0

File: app/forms/state_file/md_tax_refund_form.rb (hunk 19 above)
Comment (arinchoi03) [reply app/forms/state_file/md_tax_refund_form.rb:29]: https://codeforamerica.atlassian.net/browse/FYST-526?focusedCommentId=13466

File: app/forms/state_file/md_tax_refund_form.rb (hunk 18 above)
Comment (arinchoi03) [reply app/forms/state_file/md_tax_refund_form.rb:40]: addressed 32a3033da321cde88ea8177db00c66950639f251

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 20]
```diff
@@ -446,5 +446,57 @@
...
+      before do
+        allow_any_instance_of(Efile::Md::Md502Calculator).to receive(:calculate_authorize_direct_deposit).and_return 'X'
+      end
+
+      it 'outputs the total state and local tax withheld' do
+        expect(xml.at("Form502 AuthToDirectDepositInd")&.text).to eq('X')
+      end
+    end
+
+    context "Line 51d: NameOnBankAccount" do
```
[the remark this answers is not in the document]
Comment (arinchoi03) [reply spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:460]: see lines 501-508!

File: db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb [hunk 21]
```diff
@@ -0,0 +1,14 @@
+class AddJointAccountHolderNameAndAccountHolderDeconstructedFields < ActiveRecord::Migration[7.1]
+  def change
+    add_column :state_file_md_intakes, :has_joint_account_holder, :integer, default: 0
```
Comment (jenny-heath) [inline db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb:3]: ack i'm so sorry but we also usually have `null: false` on the enum columns. forgot to mention in the last comment 😓

File: spec/lib/efile/md/md502_calculator_spec.rb [hunk 22]
```diff
@@ -1106,4 +1106,22 @@
       expect(instance.refund_or_owed_amount).to eq(0)
     end
   end
+
+  describe "#calculate_authorize_direct_deposit" do
+    context "bank authorization was given" do
+      it "should return X" do
```
Comment (jenny-heath) [inline spec/lib/efile/md/md502_calculator_spec.rb:1112]: minor but it now returns true (same for "nil" below)

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 23]
```diff
@@ -446,5 +446,67 @@
         expect(xml.at("Form502 TaxWithheld")&.text).to eq('500')
       end
     end
+
+    context "AuthToDirectDepositInd" do
+      before do
+        intake.bank_authorization_confirmed = "yes"
```
Comment (jenny-heath) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:452]: sorry if this is a ridiculous question but does this...work? i would usually
`intake.update(bank_authorization_confirmed: "yes")`

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 24]
```diff
@@ -446,5 +446,67 @@
...
+        expect(xml.at("Form502 AuthToDirectDepositInd")&.text).to eq('X')
+      end
+    end
+
+    context "Line 51d: NameOnBankAccount" do
+      before do
+        intake.payment_or_deposit_type = "direct_deposit"
+        intake.account_holder_first_name = "Jack"
+        intake.account_holder_middle_initial = "D"
+        intake.account_holder_last_name = "Hansel"
```
Comment (jenny-heath) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:465]: same question. idk if i just completely missed this on the first review sorry

**Review by jenny-heath — APPROVED** [review]
looks pretty good! a couple of minor comments but i don't think they require re-review.

the only other thing, which can be a follow-up to this work because it's just adding ignored columns: will we lose historical data when we remove `df_data_import_failed_at` altogether? does anyone care?

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb (hunk 23 above)
Comment (arinchoi03) [reply spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:452]: yeahh i'm not sure what's going on but
`allow(intake).to receive(:bank_authorization_confirmed_yes?).and_return true` and `intake.bank_authorization_confirmed = "yes"`

passes this spec, but fails with:
`intake.bank_authorization_confirmed_yes!`
`intake.update(bank_authorization_confirmed: "yes")`

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb (hunk 24 above)
Comment (arinchoi03) [reply spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:465]: Screenshot 2024-11-25 at 7 39 20 PM

it works, it seems

File: db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb (hunk 21 above)
Comment (arinchoi03) [reply db/migrate/20241121222604_add_joint_account_holder_name_and_account_holder_deconstructed_fields.rb:3]: i should've caught that! ty

**Comment by arinchoi03** [comment]
@jenny-heath I think that's a great question about `df_data_import_failed_at` -- I did create a ticket in the story where I was attempting to ignore it https://codeforamerica.atlassian.net/browse/FYST-1173
We're not really dropping the column _yet_ but I think we should definitely discuss if we should. I'll add a computer gardening topic to follow up on this & potentially create a story if we think we want to drop with note to alert DS :)

File: app/forms/state_file/md_tax_refund_form.rb (hunk 19 above)
Comment (arinchoi03) [reply app/forms/state_file/md_tax_refund_form.rb:29]: answered! 
https://codeforamerica.atlassian.net/browse/FYST-526?focusedCommentId=13486
>@Arin Choi great catch--let’s allow alphabet, apostrophes and dash, and white space only. no numbers

File: app/forms/state_file/md_tax_refund_form.rb [hunk 25]
```diff
@@ -46,8 +46,8 @@ class MdTaxRefundForm < QuestionsForm
       with_options if: -> { has_joint_account_holder == 'yes' } do
         validates :joint_account_holder_first_name, presence: true
         validates :joint_account_holder_last_name, presence: true
-        validates :joint_account_holder_first_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,15})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.first_name.invalid_format') }}
-        validates :joint_account_holder_last_name, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,137})\z/.freeze, message: ->(_object, _data) { I18n.t('errors.attributes.last_name.invalid_format') }}
```
Comment (arinchoi03) [inline app/forms/state_file/md_tax_refund_form.rb]: rubymine said `{1}` after the `[a-zA-Z]` was redundant, so I removed it

**Comment by arinchoi03** [comment]
closing & reopening b/c migrations were changed
